// @larsire/common-logger/src/logger.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
let config = require(process.cwd() + "/config/default.json");
const fs = require("fs");
function exportToFile(message) {
    if (message) {
        let dir = process.cwd() + (config.logDirectory || process.cwd() + "logs/");
        if (!fs.existsSync(dir)) {
            fs.mkdirSync(dir);
        }
        fs.appendFileSync(dir + "log", message + '\n');
    }
}
function prepareString(data) {
    if (typeof data === "string" || typeof data === "number") {
        return data;
    }
    else {
        try {
            return JSON.stringify(data);
        }
        catch (err) {
            return data;
        }
    }
}
function getDate() {
    let date = new Date();
    return date.getDate() + "." + date.getMonth() + "." + date.getFullYear() + " " + date.getHours() + ":" + date.getMinutes();
}
function processMessage(type, color, data) {
    let date = getDate();
    let message = date + "[" + type + "]" + prepareString(data);
    console.log(color, message);
    exportToFile(message);
}
function warn(data) {
    processMessage("Warning", "\x1b[33m%s\x1b[0m", data);
}
exports.warn = warn;
;
function error(data) {
    processMessage("Error", "\x1b[31m%s\x1b[0m", data);
}
exports.error = error;
;
function info(data) {
    processMessage("Info", "\x1b[37m%s\x1b[0m", data);
}
exports.info = info;
;
function trace(data) {
    processMessage("Trace", "\x1b[36m%s\x1b[0m", data);
}
exports.trace = trace;
;
